backend/src/packages/users/user.repository.ts

```diff
@@ -70,16 +53,13 @@ class UserRepository implements Repository {
         fullName,
       },
     };
-
     const user = await this.userModel
       .query()
       .insertGraph(userData)
       .returning('*')
+      .castTo<UserWithUserDetailsJoin>()
```

Comment:
here you will need something like 
```
 const user = await this.userModel
      .query()
      .insertGraphAndFetch({
        email,
        passwordSalt,
        passwordHash,
        [UsersTableRelation.USER_DETAILS]: {
          fullName,
        },
      })
      .withGraphFetched(UsersTableRelation.USER_DETAILS)
      .execute();
```

Reply:
I just implemented it according to a comment from Volodymir. So change it to insertGraphAndFetch?

Reply:
https://bsa2023.slack.com/archives/C05P563E2MN/p1692777744994499?thread_ts=1692777116.939539&cid=C05P563E2MN

Reply:
Check please combination of insertGraphAndFetch with withGraphFetched, or just insertGraphAndFetch. If the result are equal?